### PR TITLE
Python 3 support for ParaText

### PR DESCRIPTION
**Pull request by deads**
Hi,

I have adapted the code to support Python 3. The SWIG bundled with Ubuntu 14.04 is not fully compatible with Python 3.

Conda provides a much later version of SWIG (3.0.8) that works fine in Python 3:
```
$ conda install swig
$ hash -r
$ which swig
/opt/anaconda3/bin/swig
```

Please let me know if you have any issues with it. It's near ready to merge.

Best,
Damian

### CODE REVIEW

File: python/paratext/core.py [hunk 1]
```diff
@@ -28,6 +28,10 @@
 
 import paratext_internal as pti
 
+import sys
+if sys.version_info < (3,):
+    range = xrange
```
Comment (wesm) [inline python/paratext/core.py]: `six` may be able to help with this

https://pythonhosted.org/six/#module-six.moves

File: python/paratext/core.py [hunk 2]
```diff
@@ -189,7 +193,7 @@ def internal_csv_loader_transfer(loader, forget=True, expand=False):
          same string object to save space.
 
     """
-    for i in xrange(0, loader.get_num_columns()):
+    for i in range(0, loader.get_num_columns()):
```
Comment (wesm) [inline python/paratext/core.py]: `range(0, n)` -> `range(n)`?

File: src/python/numpy_helper.hpp [hunk 3]
```diff
@@ -160,7 +165,12 @@ struct build_array_from_range_impl<Iterator, typename std::enable_if<std::is_sam
       size_t i = 0;
       for (Iterator it = range.first; it != range.second; it++, i++) {
         PyObject **ref = (PyObject **)PyArray_GETPTR1((PyArrayObject*)array, i);
-        PyObject *newobj = PyString_FromStringAndSize((*it).c_str(), (*it).size());
+        PyObject *newobj = NULL;
+        #if PY_MAJOR_VERSION < 3
+        newobj = PyString_FromStringAndSize((*it).c_str(), (*it).size());
+        #else
+        newobj = PyUnicode_FromStringAndSize((*it).c_str(), (*it).size());
+        #endif
```
Comment (wesm) [inline src/python/numpy_helper.hpp]: Put this in an inline helper method?

`PyUnicode_FromStringAndSize` assumes UTF-8 -- I'm creating an issue about other string encodings (or you can force bytes and decode after parsing)

**Comment by wesm** [comment]
lgtm

**Comment by deads** [comment]
@wesm, I've made the suggested changes. I'll go ahead and merge.

**Pull request merged**